src/View/mainpage/MainPage.py

```diff
@@ -258,8 +307,71 @@ def zoom_out(self, is_four_view):
 
     def format_data(self, size):
         """
-        This function is used to update the meta data's font size and margin based on the height and width of the
-        viewports.
+        This function is used to update the meta data's font size and margin
+        based on the height and width of the viewports.
         size: The size of the DicomStackedWidget
         """
         self.dicom_axial_view.format_metadata(size)
+
+    def create_image_fusion_tab(self):
+        """
+        This function is used to create the tab for image fusion.
+        Function checks if the moving dict container contains rtss to
+        load rtss. Views are created and stacked into three window view.
+        """
+        # Instance of Moving Model
+        moving_dict_container = MovingDictContainer()
+        if moving_dict_container.has_modality("rtss"):
+            if not hasattr(self, 'structures_tab'):
+                self.structures_tab = StructureTab(True)
+                self.structures_tab.request_update_structures.connect(
+                    self.update_views)
+                self.left_panel.addTab(self.structures_tab, "Structures")
```

Comment:
What is the purpose of lines 325-329 ? Regardless of whether a RTSTRUCT file is selected or not, the structure tab will always be visible. So, I think this chunk of code will never be run.

Reply:
> What is the purpose of lines 325-329 ? Regardless of whether a RTSTRUCT file is selected or not, the structure tab will always be visible. So, I think this chunk of code will never be run.

I've written a check to see if the current RTSTRUCT is has no ROIs and if so, load the new RTSTRUCT. Also currently if a new RTSTRUCT is loaded, it will crash if it belongs to a different image set, so I have written a check to disable new RTSTRUCTS from being loaded until this functionality is added

Reply:
StructureTab has an instance of DrawROIWindow and DeleteROIWindow. Both of these classes are inherited from QMainWindow. If you create a new instance of StructureTab (line 328), there will be memory leak because of these QMainWindows. 
I think you should update the StructureTab instead of creating a new instance of it.

Reply:
> I think you should update the StructureTab instead of creating a new instance of it.

Code has been replaced so structure_tab.update_ui() is used. the update_ui function has been changed to have a choice between loading from PatientDictContainer and MovingDictContainer